gui/wxpython/startup/guiutils.py

```diff
@@ -692,6 +696,58 @@ def delete_grassdb_interactively(guiparent, grassdb):
     return deleted
 
 
+def can_switch_mapset_interactively(guiparent, grassdb, location, mapset):
+    """
+    Checks if mapset is locked and offers to remove the lock file.
+
+    Returns True if user wants to switch to the selected mapset in spite of
+    removing lock. Returns False if a user wants to stay in the current
+    mapset or if an error was encountered.
+    """
+    can_switch = True
+
+    user = get_current_user()
+    lockfile = get_lockfile_if_present(grassdb, location, mapset)
+    timestamp = (datetime.datetime.fromtimestamp(
+        os.path.getmtime(lockfile))).replace(microsecond=0)
+    if lockfile:
+        dlg = wx.MessageDialog(
+            parent=guiparent,
+            message=_("User {user} is already running GRASS in selected mapset "
```

Comment:
This gets the current user and than it claims it is the one running the session. Well, that may or may not be true. (It is actually a more useful information when it is not true.)

It seems that the best way of obtaining it is pathlib's `Path.owner`.

Reply:
> 
> 
> This gets the current user and than it claims it is the one running the session. Well, that may or may not be true. (It is actually a more useful information when it is not true.)
> 
> It seems that the best way of obtaining it is pathlib's `Path.owner`.

I made actually the same function as get_mapset_owner, just for gislock owner. Maybe we could make one function get_file_owner instead of get_mapset_owner and get_gislock_owner.